File: hw/gudino/u2/hw19 Inheritance/Python/FarmAnimal.py
# FarmAnimal.py
class FarmAnimal:
    def __init__(self, id, breed, born_on, gender, is_able_to_reproduce, weight, cage):
        self.id = id
        self.breed = breed
        self.born_on = born_on
        self.gender = gender
        self.is_able_to_reproduce = is_able_to_reproduce
        self.weight = weight
        self.cage = cage

    def __str__(self):
        return (
            f"FarmAnimal(id={self.id}, breed={self.breed}, born_on={self.born_on}, "
            f"gender={self.gender}, ableToReproduce={self.is_able_to_reproduce}, "
            f"weight={self.weight}, cage={self.cage})"
        )

File: hw/pastor/u2/hw19 Inheritance/Javascript/ec.edu.espe.farmsystem.model/farmAnimal.js
class FarmAnimal {
    constructor(id, breed, bornOn, gender, isAbleToReproduce, weight, cage, location) {
        this.id = id;
        this.breed = breed;
        this.bornOn = bornOn;
        this.gender = gender;
        this.isAbleToReproduce = isAbleToReproduce;
        this.weight = weight;
        this.cage = cage;
        this.location = location;
    }

    getAgeInMonths() {
        const today = new Date();
        let months = (today.getFullYear() - this.bornOn.getFullYear()) * 12;
        months -= this.bornOn.getMonth();
        months += today.getMonth();
        return months <= 0 ? 0 : months;
    }

    toString() {
        const typeName = this.constructor.name.toUpperCase();
        const reproStr = this.isAbleToReproduce ? "Yes" : "No";
        const cageStr = this.cage ? this.cage.toString() : "No Cage Assigned";
        const dateStr = this.bornOn.toISOString().split('T')[0];

        return `
========================================
           FARM ANIMAL: ${typeName}
========================================
 ID              : ${this.id}
 Breed           : ${this.breed}
 Gender          : ${this.gender}
 Weight          : ${this.weight.toFixed(1)} kg
 Born On         : ${dateStr}
 Reproduces      : ${reproStr}
 Location        : ${this.location}
 Cage            : ${cageStr}`;
    }
}

module.exports = FarmAnimal;